simplecpp.cpp

```diff
@@ -2719,14 +2719,36 @@ static std::string toAbsolutePath(const std::string& path) {
     return simplecpp::simplifyPath(path);
 }
 
-static std::pair<std::string, bool> extractRelativePathFromAbsolute(const std::string& absolutepath) {
-    static const std::string prefix = currentDirectory() + "/";
-    if (startsWith_(absolutepath, prefix)) {
-        const std::size_t size = prefix.size();
-        return std::make_pair(absolutepath.substr(size, absolutepath.size() - size), true);
+static std::string dirPath(const std::string& path, bool withTrailingSlash=true) {
+    const std::size_t lastSlash = path.find_last_of("\\/");
+    if (lastSlash == std::string::npos) {
+        return "";
     }
-    // otherwise
-    return std::make_pair("", false);
+    return path.substr(0, lastSlash + (withTrailingSlash ? 1U : 0U));
+}
+
+static std::string omitPathTrailingSlash(const std::string& path) {
+    if (endsWith(path, "/")) {
+        return path.substr(0, path.size() - 1U);
+    }
+    return path;
+}
+
+static std::string extractRelativePathFromAbsolute(const std::string& absoluteSimplifiedPath, const std::string& prefixSimplifiedAbsoluteDir = currentDirectory()) {
+    const std::string normalizedAbsolutePath = omitPathTrailingSlash(absoluteSimplifiedPath);
+    std::string currentPrefix = omitPathTrailingSlash(prefixSimplifiedAbsoluteDir);
+    std::string leadingParenting;
+    while (!startsWith_(normalizedAbsolutePath, currentPrefix)) {
```

Comment:
how does this work when a relative path can't be created.. i.e. if current directory is "C:\test1" then what is the relative path for "D:\foo"

Reply:
> What it should do in this case is to return ../foo. Is it working? 

that will not work in windows. `../foo` would point at `C:\foo` if current folder is `C:\test1`.
And something like `../../d:/foo` will not work neither.

Reply:
> > What it should do in this case is to return ../foo. Is it working?
> 
> that will not work in windows. `../foo` would point at `C:\foo` if current folder is `C:\test1`. And something like `../../d:/foo` will not work neither.

Oh I missed the `D` here. Anyway, this case will never happen - since in the first place if the `#include` directive is relative it will never can break outside of the partition `C:`.

Reply:
Ideally we would have a regression test for 6ee95f7 however I'm not sure how to do that. A pytest would require that there was different drives right? or can we test that the absolute path is returned even if it does not exist?

Reply:
for your information, feel free to ping me if it takes couple of days for a response.